### PR TITLE
Liveness probe with subservices

### PR DESCRIPTION
**Pull request by zolug**
## Description
- NSP, IPAM, Proxy and LB register _Liveness_ probe service at custom gRPC health server.
- FE no longer checks Target Registry Client as part of Readiness.
- IPAM, NSP Readiness no longer includes their server status.
- Liveness probes in IPAM, NSP, Proxy and LB containers rely on service name "Liveness" to check status.
- Health server in TAPA hosts health services _Liveness_, _Startup_ and _Readiness_ out of the box. Ambassador server availability is returned as part of Liveness. (Example-target has been updated accordingly.) Note, it's not mandatory to use these health services in the kubernetes probe configuration, but using them could avoid the need of revisiting the probe configuration in the TAPA container if the recommended probe setup changes (the implementation would change in the background).

Note: There's a DR that does not recommend putting evaluation criteria inside startup probe which are related to service operation readiness which later can fail readiness. Thus startup probes do not check for server availability. (Although in case of IPAM and NSP this limitation would no longer stand.)

## Issue link
#480 

## Checklist

- Purpose
    - [ ] Bug fix
    - [x] New functionality
    - [ ] Documentation
    - [ ] Refactoring
    - [ ] CI
- Test
    - [x] Unit test
    - [ ] E2E Test
    - [x] Tested manually
- Introduce a breaking change
    - [ ] Yes (description required)
    - [x] No

### CODE REVIEW

File: pkg/health/const.go [hunk 1]
```diff
@@ -27,19 +27,26 @@ const (
...
-var FEReadinessServices []string = []string{NSPCliSvc, TargetRegistryCliSvc, EgressSvc}
+var LBLivenessServices []string = []string{NSMEndpointSvc}
+var FEReadinessServices []string = []string{NSPCliSvc, EgressSvc}
 var ProxyReadinessServices []string = []string{IPAMCliSvc, NSPCliSvc, NSMEndpointSvc, EgressSvc}
-var IPAMReadinessServices []string = []string{NSPCliSvc, IPAMSvc}
-var NSPReadinessServices []string = []string{NSPSvc}
+var ProxyLivenessServices []string = []string{NSMEndpointSvc}
+var IPAMReadinessServices []string = []string{NSPCliSvc}
+var IPAMLivenessServices []string = []string{IPAMSvc}
+var NSPReadinessServices []string = []string{}
```
Comment (LionelJouin) [inline pkg/health/const.go:48]: Why keeping the empty sub-services? (NSPReadinessServices, TAPAStartupServices and TAPAReadinessServices)

**Review by LionelJouin — COMMENTED** [review]
Could you also add the type of probes for each sub-probe/sub-service in the documentation

File: pkg/health/const.go (hunk 1 above)
Comment (zolug) [reply pkg/health/const.go:48]: Hmm, could be that these are just confusing, so I will remove them.

File: docs/components/tapa.md [hunk 2]
```diff
@@ -81,7 +84,12 @@ An overview of the communications between all components is available [here](res
 
 The health check is provided by the [GRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md). The status returned can be `UNKNOWN`, `SERVING`, `NOT_SERVING` or `SERVICE_UNKNOWN`.
 
-TODO
+Service | Description
+--- | ---
+Liveness | A unique service to be used by liveness probe to return status, can aggregate other lesser services
+Startup | A unique service to be used by startup probe to return status, can aggregate other lesser services
+Readiness | A unique service to be used by readiness probe to return status, can aggregate other lesser services
+AmbassadorSvc | Monitor status of the Ambassador server
```
[the remark this answers is not in the document]
Comment (zolug) [reply docs/components/tapa.md:92]: I don't get this split. All are health services, and it's up to the user to decide which one to use and where. Liveness/Startup/Readiness are the services which are intended to be used by probes.

File: cmd/tapa/main.go [hunk 3]
```diff
@@ -115,6 +117,16 @@ func main() {
 
 	// create and start health server
 	sigCtx = health.CreateChecker(sigCtx)
+	// add health server services Startup, Readiness and Liveness representing probes
+	if err := health.RegisterStartupSubservices(sigCtx); err != nil {
```
[the remark this answers is not in the document]
Comment (zolug) [reply cmd/tapa/main.go:121]: Their purpose is to hide implementation, thus not requiring kubernetes probe config changes in the TAPA container (avoiding/limiting user impact).

**Review by LionelJouin — APPROVED** [review]

**Pull request merged**